plugins/aws/ec2/unassociatedElasticIp.js

```diff
@@ -0,0 +1,52 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Unassociated Elastic IP Addresses',
+    category: 'EC2',
+    description: 'Ensures all EIPs are allocated to a resource to avoid accidental usage or reuse and to save costs',
+    more_info: 'EIPs should be deleted if they are not in use to avoid extra charges.',
+    recommended_action: 'Delete the unassociated Elastic IP',
+    link: 'https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html',
+    apis: ['EC2:describeAddresses'],
+    
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        async.each(regions.ec2, function(region, rcb) {
+            var describeAddresses = helpers.addSource(cache, source,
+                ['ec2', 'describeAddresses', region]);
+
+            if (!describeAddresses) return rcb();
+
+            if(describeAddresses.err || !describeAddresses.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for Elastic IP Addresses: ' + helpers.addError(describeAddresses), region);
+                return rcb();
+            }
+
+            if(!describeAddresses.data.length) {
+                helpers.addResult(results, 0, 'No Elastic IP Addresses found', region);
+                return rcb();
+            }
+
+            var unassociatedElasticIpFound = false;
+            describeAddresses.data.forEach(function(elasticIp){
+                if(!elasticIp.AssociationId) {
+                    unassociatedElasticIpFound = true;
+                    helpers.addResult(results, 2, 'Elastic IP address is not associated to any resource', region, elasticIp.AllocationId);
+                }
+            });
+            
+            if(!unassociatedElasticIpFound){
+                helpers.addResult(results, 0, 'All Elastic IP addresses are associated to resources', region);
+            }
```

Comment:
@matthewdfuller for EIP should we be returning a pass/fail for each EIP or will this be ok?

Reply:
From call: each one should have its own result